src/briefcase/platforms/linux/system.py

```diff
@@ -0,0 +1,987 @@
+            briefcase_min_version = ".".join(
+                str(v) for v in self.briefcase_required_python_version
+            )
+            raise BriefcaseCommandError(
+                f"The system python3 version provided by {app.target_image} "
+                f"is {app.python_version_tag}; Briefcase requires a "
+                f"minimum Python3 version of {briefcase_min_version}."
+            )
+        elif target_python_version != (
+            self.tools.sys.version_info.major,
+            self.tools.sys.version_info.minor,
+        ):
+            self.logger.warning(
+                f"""
+*************************************************************************
+** WARNING: Python version mismatch!                                   **
+*************************************************************************
+
+    The system python3 provided by {app.target_image} is {app.python_version_tag}.
+    This is not the same as your local system ({self.python_version_tag}).
+
+    Ensure you have tested for Python version compatibility before
+    releasing this app.
+
+*************************************************************************
+"""
+            )
+
+    def verify_system_python(self):
+        """Verify that the Python being used to run Briefcase is the
+        default system python.
+
+        Will raise an exception if the system Python isn't an obvious Python3,
+        or the Briefcase Python isn't the same version as the system Python.
+
+        Requires that the app tools have been verified.
+        """
+        system_python_bin = Path("/usr/bin/python3").resolve()
+        system_version = system_python_bin.name.split(".")
+        if system_version[0] != "python3" or len(system_version) == 1:
+            raise BriefcaseCommandError("Can't determine the system python version")
+
+        if system_version[1] != str(self.tools.sys.version_info.minor):
+            raise BriefcaseCommandError(
+                f"The version of Python being used to run Briefcase ({self.python_version_tag}) "
+                f"is not the system python3 (3.{system_version[1]})."
+            )
+
+    def _system_requirement_tools(self, app: AppConfig):
+        """Utility method returning the packages and tools needed to verify
+        system requirements.
+
+        :param app: The app being built.
+        :returns: A triple containing (0) The list of package names that must
+            be installed at a bare minimum; (1) the arguments for the command
+            used to verify the existence of a package on a system, and (2)
+            the command used to install packages. All three values are `None`
+            if the system cannot be identified.
+        """
+        if app.target_vendor_base == DEBIAN:
+            base_system_packages = [
+                "python3-dev",
+                "python3-venv",
+                "python3-pip",
+                "build-essential",
+            ]
```

Comment:
My daily driver machine errored on `python3-venv`....that made we want to review these packages more closely....especially since I can build a `deb` without any of these packages installed.

The `python3-dev` package ([focal](https://packages.ubuntu.com/focal/amd64/python3-dev/filelist), [jammy](https://packages.ubuntu.com/jammy/amd64/python3-dev/filelist), [bullseye](https://packages.debian.org/bullseye/python3-dev/filelist)) seems to contain docs and `/usr/bin/python3-config`. I don't think this is being used anywhere. However, this pulls in `libpython3-dev` which we definitely need to build the stub.

The `python3-venv` package ([focal](https://packages.ubuntu.com/focal/amd64/python3-venv/filelist), [jammy](https://packages.ubuntu.com/jammy/amd64/python3-venv/filelist), [bullseye](https://packages.debian.org/bullseye/python3-venv/filelist)) seems to contain docs and (only for focal) `/usr/bin/pyvenv`....which is a broken sym link to `/usr/bin/pyvenv-3.8`... Nonetheless, this pulls in the specific version [`python3.8-venv`](https://packages.ubuntu.com/focal/amd64/python3.8-venv) (e.g. for focal)....and this only includes [`ensurepip`](https://packages.ubuntu.com/focal/amd64/python3.8-venv/filelist). But where are we using virtual environments anyway?

The `python3-pip` package ([focal](https://packages.ubuntu.com/focal/amd64/python3-pip/filelist), [jammy](https://packages.ubuntu.com/jammy/amd64/python3-pip/filelist), [bullseye](https://packages.debian.org/bullseye/python3-pip/filelist)) does actually install `pip`....but you don't need `pip` installed for the system Python in order for Briefcase to use `pip`.

The `build-essential` package also technically isn't _required_....having `gcc`, `make`, etc installed is sufficient to build the stub.

So, I think we should remove `python3-venv` no matter what else we do.

Furthermore, though, I don't think we should _require_ users to have these specific packages installed; this is a deviation from how we've handled system requirements elsewhere. However, perhaps for the sake of some base guarantees in the user's env, it probably isn't that burdensome to ask users to install these fairly basic packages....given they have `sudo` access, that is.

Reply:
That's a good point - I was copying the definitions that are present in the Dockerfile, but we don't actually use venv anywhere (other than assuming that you're using a venv). If you've got Briefcase installed, you've got `pip` in your virtual environment, so we don't need to force that package either. `build-essential` is probably installing more than is needed, but it's a good baseline set of tools - and although the stub and the Toga demo doesn't need anything more than gcc and make, other packages that need to be binary compiled *might* have more complex build requirements.

In terms of whether this varies from behavior elsewhere - my read is that we install what we can, and ask the user to install the rest. We don't install Xcode or Visual Studio, for example - we tell the user they're missing, and tell them where/how to install them. In this case, you're going to have a bad time if you don't have the dev packages for Python, and gcc/make - and the failure mode is the "pkg-config not found" mess that is completely opaque if you don't know what is happening. We can't catch that `pkg-config` error specifically - all we see is that `pip` failed; so I'd argue ensuring some basic packages that we can reliably expect to be requirements isn't overly burdensome.